dev-client/src/screens/LocationScreens/LocationSoilIdScreen.tsx

```diff
@@ -42,19 +43,21 @@ export const LocationSoilIdScreen = ({siteId, coords}: Props) => {
   );
 
   return (
-    <ScreenScaffold
-      AppBar={
-        <AppBar title={site?.name ?? t('site.dashboard.default_title')} />
-      }>
-      <SoilIdDescriptionSection siteId={siteId} />
-      <SoilIdMatchesSection siteId={siteId} />
-      {siteId ? (
-        <SiteDataSection />
-      ) : (
-        <Box paddingVertical="md">
-          <CreateSiteButton coords={coords} />
-        </Box>
-      )}
-    </ScreenScaffold>
+    <BottomSheetModalProvider>
```

Comment:
We should figure out a known 'correct' configuration for providing modal / bottom sheet support -- some screens have one of these and some don't, and they aren't consistent about where in the hierarchy it is either. There's one in the application root as well, but it's outside of the root navigator and so we can't display e.g. the "create site here" button in the outermost modal provider because it doesn't have access to the navigation context. Even with this, the header-height provider doesn't work as expected for nested sheets.

Reply:
commenting for posterity that in a convo earlier today we said that we think the correct location for `BottomSheetModalProvider` will end up being inside the screen scaffold, nested underneath the header height provider and above the safe area view. this was already an issue before the PR though so i don't expect it needs to be fixed in this PR, but feel free!

Reply:
Tracked in an updated ticket: https://github.com/techmatters/terraso-mobile-client/issues/697